core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -379,6 +380,16 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
 
   try {
     _conf = config.clone()
+
+    // if a secret key has been set in an env variable, use it
+    for (ns <- Seq(Namespace.Blank, Namespace.Application, Namespace.Submission)) {
+      sys.env.get(ns.inEnv(ENV_AUTH_SECRET))
```

Comment:
Now that I see the methods being used, the names sound kinda weird. I'd prefer `ns.envName` and `ns.confName`.